src/ngraph/runtime/reference/non_zero.hpp

```diff
@@ -0,0 +1,146 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <cstddef>
+#include "ngraph/shape.hpp"
+
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            /// \brief Return number of non-zero entries in the input argument.
+            ///
+            /// \param arg Input tensor
+            /// \param arg_shape Input tensor shape
+            /// Output number of non-zero entries in arg
+            template <typename T>
+            size_t non_zero_get_count(const T* arg, const Shape& arg_shape)
```

Comment:
@yimeisun123 if we break the non_zero op into 2 calls, will that work for any user trying to call non_zero reference kernel directly? Or is it that the interpreter will be the sole user for this op?

Reply:
Any caller can call. As with other reference kernels, the caller needs to allocate sufficient memory for output.